go-controller/pkg/ovn/secondary_layer2_network_controller.go

```diff
@@ -552,6 +556,73 @@ func (oc *SecondaryLayer2NetworkController) addUpdateLocalNodeEvent(node *corev1
 	return err
 }
 
+func (oc *SecondaryLayer2NetworkController) addUpdateRemoteNodeEvent(node *corev1.Node, syncZoneIC bool) error {
+	var errs []error
+
+	if util.IsNetworkSegmentationSupportEnabled() && oc.IsPrimaryNetwork() {
+		if syncZoneIC && config.OVNKubernetesFeature.EnableInterconnect {
+			if err := oc.addPortForRemoteNodeGR(node); err != nil {
+				err = fmt.Errorf("failed to add the remote zone node [%s] to the zone interconnect handler, err : %v", node.Name, err)
+				errs = append(errs, err)
+				oc.syncZoneICFailed.Store(node.Name, true)
+			} else {
+				oc.syncZoneICFailed.Delete(node.Name)
+			}
+		}
+	}
+
+	errs = append(errs, oc.BaseSecondaryLayer2NetworkController.addUpdateRemoteNodeEvent(node))
+
+	err := utilerrors.Join(errs...)
+	if err != nil {
+		oc.recordNodeErrorEvent(node, err)
+	}
+	return err
+}
+
+func (oc *SecondaryLayer2NetworkController) addPortForRemoteNodeGR(node *corev1.Node) error {
+	nodeJoinSubnetIPs, err := util.ParseNodeGatewayRouterJoinAddrs(node, oc.GetNetworkName())
+	if err != nil || len(nodeJoinSubnetIPs) == 0 {
+		return fmt.Errorf("failed to get the node %s join subnet IPs : %w", node.Name, err)
+	}
+
+	remoteGRPortMac := util.IPAddrToHWAddr(nodeJoinSubnetIPs[0].IP)
+	var remoteGRPortNetworks []string
+	for _, ip := range nodeJoinSubnetIPs {
+		remoteGRPortNetworks = append(remoteGRPortNetworks, ip.String())
+	}
+
+	remotePortAddr := remoteGRPortMac.String()
+	for _, tsNetwork := range remoteGRPortNetworks {
+		remotePortAddr = remotePortAddr + " " + tsNetwork
+	}
+	logicalSwitchPort := nbdb.LogicalSwitchPort{
+		Name:      types.SwitchToRouterPrefix + oc.GetNetworkScopedSwitchName(types.OVNLayer2Switch) + "_" + node.Name,
+		Type:      "remote",
+		Addresses: []string{remotePortAddr},
+	}
+	logicalSwitchPort.ExternalIDs = map[string]string{
+		types.NetworkExternalID:  oc.GetNetworkName(),
+		types.TopologyExternalID: oc.TopologyType(),
+		"node":                   node.Name,
+	}
+	nodeID := util.GetNodeID(node)
+	if nodeID == -1 {
+		// Don't consider this node as cluster-manager has not allocated node id yet.
+		return fmt.Errorf("failed to get node id for node - %s", node.Name)
+	}
+	logicalSwitchPort.Options = map[string]string{
+		"requested-tnl-key": strconv.Itoa(types.MaxLogicalPortTunnelKey - nodeID),
```

Comment:
spoke to @jcaamano / @qinqon last week about this.
the easiest way we decided is to use the same pod tunnelID allocator for l2 networks from CM for also the nodes and then annotate the node with the value and use that from here